File: java-prov/src/main/java/org/example/ProvRelationWithoutId.java
package org.example;

import org.openprovenance.prov.interop.InteropFramework;
import org.openprovenance.prov.model.Namespace;
import org.openprovenance.prov.model.ProvFactory;
import org.openprovenance.prov.vanilla.Document;


public class ProvRelationWithoutId implements TestCase {

    public void serialize(String format) {
        //TODO
    }

    public void deserialize(String format) {
        var inf = new InteropFramework();
        var document = inf.readDocumentFromFile(String.format("data/prov_relation_without_id.%s", format));
    }
}

File: python-prov/src/prov_relation_without_id.py
import prov.dot
from prov.model import ProvDocument
import prov.graph

def perform():
    document = ProvDocument()
    ns = document.add_namespace("ex", "https://example.org/")
    e = document.entity("ex:e")
    a = document.activity("ex:a")
    document.wasGeneratedBy(e,a)
    document.serialize(r"..\..\java-prov\data\prov_relation_without_id.json", format="json", indent=2)
    print(document.get_provn())